### PR TITLE
Move the Constrainedtext to coreui

### PR DESCRIPTION
**Pull request by ChengYanJin**
**Component**: Constrained text

**Description**:
The constrained text should locate in core-ui rather than the application.
So that we can reuse it inside any component to display the long text when there is no enough space for it, e.g. Table
image

**Design**:
image

### CODE REVIEW

**Comment by nicolas2bert** [comment]
Should we give developers the ability to modify the width of the tooltip box? We could also add this logic to the tooltip component.

File: src/lib/components/constrainedtext/Constrainedtext.component.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+//@flow
+import React from "react";
+import type { Node } from "react";
+import styled from "styled-components";
+import Tooltip from "../tooltip/Tooltip.component.js";
+import type { Props as TooltipProps } from "../tooltip/Tooltip.component.js";
+type Props = {
+  text: string,
+  tooltipstyle?: $PropertyType<TooltipProps, "overlaystyle">,
+  tooltipplacement?: $PropertyType<TooltipProps, "placement">,
```
Comment (JBWatenbergScality) [inline src/lib/components/constrainedtext/Constrainedtext.component.js]: use camelCase here 

```suggestion
  tooltipStyle?: $PropertyType<TooltipProps, "overlaystyle">,
  tooltipPlacement?: $PropertyType<TooltipProps, "placement">,
```

File: src/lib/components/constrainedtext/Constrainedtext.component.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+const BlockTooltip = styled.div`
+  & > .sc-tooltip {
+    display: block;
+  }
+`;
+
+function ConstrainedText({
+  text,
+  tooltipstyle,
+  tooltipplacement,
```
Comment (JBWatenbergScality) [inline src/lib/components/constrainedtext/Constrainedtext.component.js]:
```suggestion
  tooltipStyle,
  tooltipPlacement,
```

File: src/lib/components/constrainedtext/Constrainedtext.component.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+}: Props): Node {
+  return (
+    <BlockTooltip>
+      <Tooltip
+        overlay={text}
+        overlaystyle={tooltipstyle}
+        placement={tooltipplacement}
```
Comment (JBWatenbergScality) [inline src/lib/components/constrainedtext/Constrainedtext.component.js]:
```suggestion
        overlaystyle={tooltipStyle}
        placement={tooltipPlacement}
```

File: src/lib/components/constrainedtext/Constrainedtext.component.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import styled from "styled-components";
+import Tooltip from "../tooltip/Tooltip.component.js";
+import type { Props as TooltipProps } from "../tooltip/Tooltip.component.js";
+type Props = {
+  text: string,
+  tooltipstyle?: $PropertyType<TooltipProps, "overlaystyle">,
+  tooltipplacement?: $PropertyType<TooltipProps, "placement">,
+};
+
+const ConstrainedtextContainer = styled.div`
```
Comment (JBWatenbergScality) [inline src/lib/components/constrainedtext/Constrainedtext.component.js]:
```suggestion
const ConstrainedTextContainer = styled.div`
```

File: src/lib/components/constrainedtext/Constrainedtext.component.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  return (
+    <BlockTooltip>
+      <Tooltip
+        overlay={text}
+        overlaystyle={tooltipstyle}
+        placement={tooltipplacement}
+      >
+        <ConstrainedtextContainer className="sc-constrainedtext">
+          {text}
+        </ConstrainedtextContainer>
```
Comment (JBWatenbergScality) [inline src/lib/components/constrainedtext/Constrainedtext.component.js]:
```suggestion
        <ConstrainedTextContainer className="sc-constrainedtext">
          {text}
        </ConstrainedTextContainer>
```

**Review by JBWatenbergScality — APPROVED** [review]

**Review by nicolas2bert — APPROVED** [review]

**Pull request merged**